### PR TITLE
Add documentation and support for Singularity

### PR DESCRIPTION
**Pull request by nilsdeppe**
## Proposed changes

Singularity is a container alternative to Docker and can use Docker
containers. It integrates more readily with the host OS and does not require
root privileges to get set up.

The documentation is built and will be available at: http://gh.nilsdeppe.com/spectre/pr_701

Breaks:
- Travis Dockerfile, every PR needs to be rebased

### Types of changes:

- [ ] Bugfix
- [x] New feature

### Component:

- [ ] Code
- [x] Documentation
- [ ] Build system
- [ ] Continuous integration

### Code review checklist

- Follows [code review guidelines](https://sxs-collaboration.github.io/spectre/code_review_guide.html)
- Code has documentation and unit tests
- Private member variables have a trailing underscore
- Do not use [Hungarian notation](https://en.wikipedia.org/wiki/Hungarian_notation), e.g. `double* pd_blah` is bad
- Header order:
  1. hpp corresponding to cpp (only in cpp files) or `tests/Unit/TestingFramework.hpp` (only in tests)
  2. Blank line (only in cpp files and tests)
  3. STL and externals (in alphabetical order)
  4. Blank line
  5. SpECTRE includes (in alphabetical order)
- File lists in CMake are alphabetical
- Correct `noexcept` specification for functions (if unsure, mark `noexcept`)
- Mark objects `const` whenever possible
- Almost always `auto`, except with expression templates, i.e. `DataVector`
- All commits for performance changes provide quantitative evidence and the tests used to obtain said evidence.
- Make sure error messages are helpful, e.g. "The number of grid points in the matrix 'F' is not the same as the number of grid points in the determinant."
- Prefix commits addressing PR requests with `fixup`. These commits are flagged
  by Travis so we do not accidentally merge them.
- Include what you use, prefer forward declarations in hpp files.
- Explicitly make numbers floating point, e.g. `2.` or `2.0` over `2`
- Use `Tensor`'s non-member get if the indices are constant expressions, e.g.
  `get<1>(tensor)`

### CODE REVIEW

File: docs/Installation/Installation.md [hunk 1]
```diff
@@ -131,6 +131,48 @@ Notes:
...
+[Singularity](http://singularity.lbl.gov/index.html) is a container alternative
+to Docker with better security and nicer integration.
+
+To use Singularity you must:
+
+1. Build [Singularity](http://singularity.lbl.gov/index.html) and add it to your
+   `$PATH`
+2. `cd` to the directory where you want to store the SpECTRE Singularity image,
+   source, and build directories, let's call it WORKDIR, which must be somewhere
+   in your home directory. Clone SpECTRE into `WORKDIR/SPECTRE_ROOT`.
```
Comment (markscheel) [inline docs/Installation/Installation.md]: Does it really need to be in your home directory? Or just someplace user-writeable?

File: docs/Installation/Installation.md [hunk 2]
```diff
@@ -131,6 +131,48 @@ Notes:
...
+3. Run `singularity build spectre.img
```
Comment (markscheel) [inline docs/Installation/Installation.md]: When I do the singularity build command it says

ERROR UNAUTHORIZED: authentication required
ERROR Check existence, naming, and permissions

**Review by markscheel — CHANGES_REQUESTED** [review]

File: docs/Installation/Installation.md [hunk 3]
```diff
@@ -131,6 +131,48 @@ Notes:
...
+   docker://sxs-collaboration/spectrebuildenv:latest`.
+4. To start the container run `singularity shell spectre.img` and you
+   will be dropped into a bash shell. The first thing you will always need to do
+   when entering the container is:
+   `. SPECTRE_HOME/containers/SingularityLoad.sh`
+5. To set up Charm++ run `mv /work/charm ./ && cd ./charm`
```
Comment (markscheel) [inline docs/Installation/Installation.md]: Should this be /PATH/TO/CHARM (which is used below)?

File: docs/Installation/Installation.md [hunk 4]
```diff
@@ -131,6 +129,53 @@ Notes:
...
+  losing any data, needing to clean up old containers, or sharing data between
+  containers and the host.
+- To build with clang, the CMake command is:
+```
+    cmake -D CMAKE_CXX_COMPILER=clang++ \
+          -D CMAKE_C_COMPILER=clang \
+          -D CMAKE_Fortran_COMPILER=gfortran \
+          -D CHARM_ROOT=/work/charm/multicore-linux64-clang SPECTRE_ROOT
+```
+- You can safely ignore any warnings about the locale settings.
```
Comment (markscheel) [inline docs/Installation/Installation.md]: If locale warnings are gone you don't need this anymore.

**Review by markscheel — CHANGES_REQUESTED** [review]

**Review by kidder — DISMISSED** [review]

**Comment by nilsdeppe** [comment]
@markscheel can you look at this? The only thing missing is pushing the new image, but that'll break all currently travis builds...

**Review by markscheel — DISMISSED** [review]

**Review by kidder — APPROVED** [review]

**Review by wthrowe — APPROVED** [review]

**Pull request merged**